### PR TITLE
satellite/contact: add PingMe endpoint

### PR DESCRIPTION
**Pull request by andriikotko**
Change-Id: I832a72fafeacf76ad64a0129bcc6582cc4f9290d

What: add PingMe endpoint cherry-pick

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

**Review by ihaid — DISMISSED** [review]

**Review by egonelbre — CHANGES_REQUESTED** [review]
This may stall storagenode startup even though it works fine.

**Review by egonelbre — COMMENTED** [review]
Ah, sorry, got confused. The satellite side of the implementation is fine.

File: satellite/contact/endpoint.go [hunk 1]
```diff
@@ -145,3 +139,60 @@ func (endpoint *Endpoint) GetTime(ctx context.Context, req *pb.GetTimeRequest) (
...
+
+	case pb.NodeTransport_QUIC_GRPC:
+		err = endpoint.service.pingNodeQUIC(ctx, nodeURL)
+		if err != nil {
+			return nil, endpoint.checkPingRPCErr(err, nodeURL)
+		}
+		return &pb.PingMeResponse{}, nil
+
+	case pb.NodeTransport_TCP_TLS_GRPC:
+		client, err := dialNodeURL(ctx, endpoint.service.dialer, nodeURL)
```
Comment (egonelbre) [inline satellite/contact/endpoint.go:176]: connection leak

**Review by egonelbre — CHANGES_REQUESTED** [review]

File: satellite/contact/endpoint.go (hunk 1 above)
Comment (profclems) [reply satellite/contact/endpoint.go:176]: Fix here: https://review.dev.storj.io/c/storj/storj/+/6700

**Review by profclems — APPROVED** [review]

**Review by egonelbre — APPROVED** [review]

**Pull request merged**